src/Symfony/Component/Messenger/Retry/ExponentialRetryStrategy.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Retry;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Stamp\RetryCountStamp;
+
+/**
+ * The retry mechanism is based on a truncated exponential backoff algorithm.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ *
+ * @experimental in 4.3
+ * @final
+ */
+class ExponentialRetryStrategy implements RetryStrategyInterface
+{
+    private $max;
+    private $offset;
+
+    /**
+     * @param int $max    The maximum number of time to retry (0 means indefinitely)
+     * @param int $offset The offset for the first power of 2
+     */
+    public function __construct(int $max = 3, int $offset = 0)
+    {
+        $this->max = $max;
+        $this->offset = $offset;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRetryable(Envelope $message): bool
+    {
+        if (0 === $this->max) {
+            return true;
+        }
+
+        $retries = $this->getCurrentRetryCount($message);
+
+        return $retries < $this->max;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getWaitingTime(Envelope $message): int
+    {
+        $retries = $this->getCurrentRetryCount($message) + 1;
+
+        return pow(2, $retries + $this->offset) * 1000;
```

Comment:
With offset: -2 and retries: 3 
Then you will wait 0.5s, 1s, 2s. 

The default offset: 0 and retries: 3
Then you will wait 2s, 4s, 8s.

Reply:
Correct. Is that a problem? I wasn't really thinking about negative offsets... but it still seems to work ok.

Reply:
No. This is fine. At first I thought 2, 4, 8 could be too much of delay.   I just wanted to try different options and I shared the results.